src/Profiles.ts

```diff
@@ -336,6 +338,178 @@ export class Profiles {
+                zosmfProfile = allProfiles.filter((temprofile) => temprofile.name === sesName)[0];
+            }
+        } else {
+            vscode.window.showInformationMessage(localize("deleteProfile.noProfilesLoaded", "No profiles available"));
+            return;
+        }
+
+        return zosmfProfile;
+    }
+
+    public async deletePrompt(deletedProfile: IProfileLoaded) {
+        const profileName = deletedProfile.name;
+        this.log.debug(localize("deleteProfile.log.debug", "Deleting profile ") + profileName);
+        const quickPickOptions: vscode.QuickPickOptions = {
+            placeHolder: localize("deleteProfile.quickPickOption", "Are you sure you want to permanently delete ") + profileName,
+            ignoreFocusOut: true,
+            canPickMany: false
+        };
+        // confirm that the user really wants to delete
+        if (await vscode.window.showQuickPick([localize("deleteProfile.showQuickPick.yes", "Yes"),
+            localize("deleteProfile.showQuickPick.no", "No")], quickPickOptions) !== localize("deleteProfile.showQuickPick.yes", "Yes")) {
+            this.log.debug(localize("deleteProfile.showQuickPick.log.debug", "User picked no. Cancelling delete of profile"));
+            return;
+        }
+
+        const profileType = ZoweExplorerApiRegister.getMvsApi(deletedProfile).getProfileTypeName();
+        try {
+            this.deleteProf(deletedProfile, profileName, profileType);
+        } catch (error) {
+            this.log.error(localize("deleteProfile.delete.log.error", "Error encountered when deleting profile! ") + JSON.stringify(error));
+            await errorHandling(error, profileName, error.message);
+            throw error;
+        }
+
+        vscode.window.showInformationMessage("Profile " + profileName + " was deleted.");
+        return profileName;
+    }
+
+    public async deleteProf(ProfileInfo, ProfileName, ProfileType) {
+        let zosmfProfile: IProfile;
+        try {
+            zosmfProfile = await (await this.getCliProfileManager(ProfileType))
+            .delete({ profile: ProfileInfo, name: ProfileName, type: ProfileType });
+        } catch (error) {
+            vscode.window.showErrorMessage(error.message);
+        }
+        return zosmfProfile.profile;
+    }
+
+    public async deleteProfile(
+        datasetTree: IZoweTree<IZoweDatasetTreeNode>, ussTree: IZoweTree<IZoweUSSTreeNode>,
+        jobsProvider: IZoweTree<IZoweJobTreeNode>, node?: IZoweNodeType) {
+
+            let deleteLabel: string;
+            let deletedProfile: IProfileLoaded;
+            if (!node){
+                deletedProfile = await this.getDeleteProfile();
+                if (!deletedProfile) {
+                    return;
+                } else {
+                    deleteLabel = deletedProfile.name;
+                }
+            } else {
+                deletedProfile = node.getProfile();
+                deleteLabel = node.getProfileName();
+            }
```

Comment:
Would this work as well? (fewer iffs)
```suggestion
            if (!node){
                deletedProfile = await this.getDeleteProfile();
            } else {
                deletedProfile = node.getProfile();
            }
            if (!deletedProfile) {
                return;
            }
            deleteLabel = deletedProfile.name;
```